### ISSUE TITLE
Sub-zero choose wrong second language 

### ISSUE THREAD

**Issue by minihessel** [pannal/Sub-Zero.bundle#500]
I have second language set to Norwegian, but still it’s a hit and miss on one season I have. In the folder it’s aviable danish, Norwegian and Swedish subs, on many episodes it chooses Norwegian, but on some danish?

**Comment by pannal** [comment]
What chooses Norwegian? SZ doesn't choose anything for your playback defaults, if you're talking about that.

**Comment by minihessel** [comment]
Norwegian subs?

2018-04-18 16:09 GMT+02:00 pannal <notifications@github.com>:

**Comment by minihessel** [comment]
Here's a print to show you what I mean: 

<img width="454" alt="screen shot 2018-04-18 at 18 52 04" src="https://user-images.githubusercontent.com/5398823/38946400-9e50085c-4339-11e8-8268-1db4955cf1dc.png">
![Uploading Screen Shot 2018-04-18 at 18.51.53.png…]()

**Comment by ukdtom** [comment]
My guess:
The Danish ones are embedded, that get's extracted from the media by Sub-Zero, and then altered by SZ according to your settings, like "Remove HI" etc.

@pannal : If above, then maybe SZ should only extract subs from media, if they are in a lang. set by the prefs?

And just guessing here, haven't looked into the code ;)

**Comment by ukdtom** [comment]
Ohhh.....And why Danish is selected as default, must be a PMS issue

**Comment by minihessel** [comment]
All of them are embedded in the file.

tor. 19. apr. 2018 kl. 08:33 skrev Tommy Mikkelsen <notifications@github.com

**Comment by minihessel** [comment]
![651219f8-36b0-4686-a127-aa250e599b63](https://user-images.githubusercontent.com/5398823/38975677-071e37d0-43af-11e8-9f81-3bc364f151b0.jpeg)


So if this is of in PMS, you are still saying that sub-zero doesn’t select the subs? In that case, PMS should add a secondary language option here, like sub zero.

**Comment by ukdtom** [comment]
You are asking SZ to alter your media files?
Like altering an mkv file in order to delete embedded subs?

That's IMHO a very bad idea!

**Comment by minihessel** [comment]
No. I store sub's as seperate files in the folders. Sub-zero could clean
those subtitles in the folders.

2018-04-19 11:31 GMT+02:00 Tommy Mikkelsen <notifications@github.com>:

**Comment by minihessel** [comment]
Good suggestion!

tor. 19. apr. 2018 kl. 14:23 skrev Tommy Mikkelsen <notifications@github.com

**Comment by pannal** [comment]
> So if this is off in PMS, you are still saying that sub-zero doesn’t select the subs? In that case, PMS should add a secondary language option here, like sub zero.

Yes, SZ doesn't alter (and can't alter) your subtitle selection.

> And since that never will happen, how about adding a feature for deleting subs that aren’t either of the languages selected in sub zero @pannal? That would be great anyway, to clean away subs in languages you don’t need.

Yep, that's a possible feature there.

**Comment by minihessel** [comment]
I would love that feature.

**Comment by minihessel** [comment]
Also, why does Sub-zero download subs in both language 1 and 2, even though there is multiple subs stored in the mkv? Shouldn't sub-zero skip elements which has subs?(internal)

**Comment by pannal** [comment]
If that setting is enabled, yes it should. I'd need logs to verify that.

**Comment by minihessel** [comment]
<img width="565" alt="screen shot 2018-05-25 at 15 51 10" src="https://user-images.githubusercontent.com/5398823/40547920-7f5298f4-6033-11e8-9bdb-bb0378b91cc1.png">


Will the feature to clean up subs in different language(mentioned above here), be released in the next release?

**Comment by pannal** [comment]
Not sure, my time is limited right now. 
As for the embedded issue: I'd need DEBUG logs for a particular item.

**Comment by minihessel** [comment]
I thought I would save you some time, so I have found out why sub-zero chooses the wrong sub. It's because the subs names are in their local language. So not Norwegian, but Norsk (which is how we write Norwegian). Therefor sub-zero does not understand that Norsk is Norwegian. This should be added.


![image](https://user-images.githubusercontent.com/5398823/40659195-d1f793e4-634d-11e8-95fd-13c07e5ae49e.png)

**Comment by pannal** [comment]
Hmm no, that's most likely not the issue. The languages you see in the Plex interface are translated based on their language code. SZ just uses a different translation for the same language code.

**Comment by minihessel** [comment]
Okay, so just clean up everything except the "norwegian(norsk)" one then,
and don't download external subs, then it should be okay?

2018-05-29 15:16 GMT+02:00 pannal <notifications@github.com>:

**Comment by pannal** [comment]
I can add an option to clean unwanted language SRTs. But that won't solve your problem per-se, as the first three in your screenshot will remain, as they're embedded and SZ doesn't alter your media files.
